firefox_ui_tests/remote/security/test_untrusted_connection_error_page.py

```diff
@@ -0,0 +1,35 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette import By, MarionetteException
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestUntrustedConnectionErrorPage(FirefoxTestCase):
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.test_url = 'https://ssl-selfsigned.mozqa.com'
+
+    def tearDown(self):
+        FirefoxTestCase.tearDown(self)
+
+    def test_untrusted_connection_error_page(self):
+        # Test the GetMeOutOfHereButton from an Untrusted Error page
+        with self.marionette.using_context('content'):
+            self.assertRaises(
+                MarionetteException,
+                self.marionette.navigate,
+                self.test_url
+                )
+            # Wait about 1 second for about:error page to work
+            time.sleep(1)
+            button = self.marionette.find_element(By.ID,
+                                                  "getMeOutOfHereButton")
+            self.assertTrue(button.is_displayed())
+            button.click()
+            self.assertEqual(self.marionette.get_url(), 'about:home')
```

Comment:
I'm a bit worried here due to 2 reasons:

1. We do not wait until the final page has been fully loaded. We directly check the URL.
2. We hardcode the URL we compare with but do not retrieve the default homepage.

The second issue can be fixed easily by reimplementing our `getDefaultHomepage` method. Maybe best to add this as a property to the BrowserWindow class under `ui/windows.py:BrowserWindow`.